bundler.py

```diff
@@ -0,0 +1,194 @@
+# ARGUMENT PARSING
+
+# Store the parsed arguments here.
+global args
+
+parser = argparse.ArgumentParser(
+    description="Helps you spin up an op-stack rollup.")
+
+subparsers = parser.add_subparsers(
+    title="commands",
+    dest="command",
+    metavar="<command>")
+
+subparsers.add_parser(
+    "start",
+    help="start an ERC4337 bundler")
+
+subparsers.add_parser(
+    "clean",
+    help="cleanup bundler processes")
+
+parser.add_argument(
+    "--no-ansi-esc",
+    help="disable ANSI escape codes for terminal manipulation",
+    default=True,
+    dest="use_ansi_esc",
+    action="store_false")
+
+####################################################################################################
+# SETUP
+
+def start():
+    setup_4337_contracts()
+    setup_stackup_bundler()
+    setup_paymaster()
+
+# --------------------------------------------------------------------------------------------------
+
+def setup_4337_contracts():
+    github_url = "https://github.com/0xFableOrg/account-abstraction.git"
+
+    if os.path.isfile("account-abstraction"):
+        raise Exception("Error: 'account-abstraction' exists as a file and not a directory.")
+    elif not os.path.exists("account-abstraction"):
+        descr = "clone the account-abstraction repository"
+        lib.run(descr, f"git clone {github_url}")
+        print(f"Succeeded: {descr}")
+
+    # If contracts have not been previously deployed
+    if not os.path.exists("account-abstraction/deployments/opstack"):
+        log_file = "logs/build_4337_contracts.log"
+        lib.run_roll_log(
+            "install account abstraction dependencies", 
+            command=deps.cmd_with_node("yarn install"), 
+            cwd="account-abstraction",
+            log_file=log_file
+        )
+        # we need to configure the network to point to local network
+        # we also need to set private key for deployment
+        priv_key = input("Enter private key that you would like to deploy contracts with: ")
+        lib.run("set private key", f"echo PRIVATE_KEY={priv_key} > account-abstraction/.env")
+        log_file = "logs/deploy_4337_contracts.log"
+        lib.run_roll_log(
+            "deploy contracts", 
+            command=deps.cmd_with_node("yarn deploy --network opstack"), 
+            cwd="account-abstraction",
+            log_file=log_file
+        )
+        print("Account abstraction contracts successfully deployed.")
+    else:
+        print("Account abstraction contracts already deployed.")
+
+def setup_stackup_bundler():
+    github_url = "github.com/stackup-wallet/stackup-bundler"
+    version = "latest"
+
+    lib.run("installing stackup bundler", f"go install {github_url}@{version}")
+    print("Installation successful")
+
+    # set environment variables for bundler
+    lib.run("set full node RPC", "echo ERC4337_BUNDLER_ETH_CLIENT_URL=http://localhost:8545 > .env")
```

Comment:
You probably want to use L2Config to get the RPC address.

Reply:
got it.